src/api/deploy/aws-elastic.ts

```diff
@@ -0,0 +1,267 @@
+    let s3Bucket = preferences.s3Bucket;
+    let codeCommit = preferences.codeCommit;
+    let applicationType = preferences.applicationType;
+    let environmentName = preferences.environmentName;
+    let solutionStackName = preferences.solutionStackName;
+    let environmentTemplate = preferences.environmentTemplate;
+    let errors = false;
+
+    if (!application) {
+      const msg = chalk.red('application is not set in yml config file \r\n');
+      observer.next({ type: 'data', data: msg});
+      errors = true;
+    }
+
+    if (!environmentName) {
+      const msg = chalk.red('environmentName is not set in yml config file \r\n');
+      observer.next({ type: 'data', data: msg});
+      errors = true;
+    }
+
+    if (!accessKeyId) {
+      if (preferences && preferences.accessKeyId) {
+        accessKeyId = preferences.accessKeyId;
+      } else {
+        const msg = chalk.red('accessKeyId is not set in environment '
+          + 'variables or in yml config \r\n');
+        observer.next({ type: 'data', data: msg});
+        errors = true;
+      }
+    }
+
+    if (!secretAccessKey) {
+      if (preferences && preferences.secretAccessKey) {
+        secretAccessKey = preferences.secretAccessKey;
+      } else {
+        const msg = chalk.red('secretAccessKey is not set in environment variables or'
+          + ' in yml config \r\n');
+        observer.next({ type: 'data', data: msg});
+        errors = true;
+      }
+    }
+
+    if (!region) {
+      if (preferences && preferences.region) {
+        region = preferences.region;
+      } else {
+        const msg =
+          chalk.red('region is not set in environment variables or in yml config file \r\n');
+        observer.next({ type: 'data', data: msg});
+        errors = true;
+      }
+    }
+
+    let sourceRepository;
+    let sourceLocation;
+    if (s3Bucket && codeCommit) {
+      const msg = chalk.red('Specify a source bundle in S3 bucket or a commit in an AWS CodeCommit'
+        + 'repository, but not both. \r\n');
+      observer.next({ type: 'data', data: msg});
+      errors = true;
+    } else if (s3Bucket) {
+      sourceRepository = 'S3';
+      sourceLocation = s3Bucket;
+    } else if (codeCommit) {
+      sourceRepository = 'CodeCommit';
+      sourceLocation = codeCommit;
+    }
+
+    if (!errors) {
+      if (!version) {
+        const date = (new Date()).toLocaleDateString();
+        version = `abstruse_${date}`;
+      }
+
+      if (!description) {
+        description = 'Deployed with Abstruse.';
+      }
+
+      if (!applicationType) {
+        applicationType = 'zip';
+      }
+
+      let msg = style.yellow.open + style.bold.open + '==> deploy started' +
+      style.bold.close + style.yellow.close + '\r\n';
+      observer.next({ type: 'data', data: msg });
+
+      // 2. install awscli and set credentials
+      let command = { type: CommandType.deploy, command: 'sudo apt-get install awscli -y' };
```

Comment:
is it a good thing to hardcode `sudo apt-get`? `apt-get` is included in debian-like Linux OS-es, but what if you use alpine for example?